R/init.R

```diff
@@ -122,52 +121,59 @@ init <- function(data,
       )
     )
   }
-
+  if (is.list(data) && !inherits(data, "teal_data_module")) {
+    checkmate::assert_list(data, names = "named")
+    data <- do.call(teal.data::teal_data, data)
+  }
   checkmate::assert_multi_class(data, c("teal_data", "teal_data_module"))
-  checkmate::assert_multi_class(modules, c("teal_module", "list", "teal_modules"))
+
+  ## `modules`
+  if (inherits(modules, "teal_module")) {
+    modules <- list(modules)
+  }
+  if (inherits(modules, "list")) {
+    modules <- do.call(teal::modules, modules)
+  }
+  checkmate::assert_class(modules, "teal_modules")
+
+  ## `filter`
   checkmate::assert(
     checkmate::check_class(filter, "teal_slices"),
     checkmate::check_list(filter, names = "named")
   )
+
+  ## all other arguments
   checkmate::assert_multi_class(title, c("shiny.tag", "character"))
   checkmate::assert_multi_class(header, c("shiny.tag", "character"))
   checkmate::assert_multi_class(footer, c("shiny.tag", "character"))
   checkmate::assert_character(id, max.len = 1, any.missing = FALSE)
```

Comment:
Shouldn't this be `assert_string`?

Reply:
There is a difference for `character(0)` - string will raise whereas character with max.len = 1 will not. `assert_string` does not have appropriate argument to allow `character(0)`. Looking at the argument default value (i.e. `character(0)`) I think this needs to stay as is

Reply:
Oh, sorry, I neglected to look at the default.